samanthas_telegram_bot/api_clients/backend/backend_client.py

```diff
@@ -0,0 +1,364 @@
+    def _get_url_and_data_for_person_creation(
+        update: Update,
+        context: CUSTOM_CONTEXT_TYPES,
+        personal_info_id: int,
+    ) -> tuple[str, DataDict]:
+        """Returns url and data for a POST request to create a person, based on user data."""
+        user_data = context.user_data
+        if user_data.role not in (Role.STUDENT, Role.TEACHER):
+            raise NotImplementedError(
+                f"Cannot produce url and data to post: {user_data.role=} not supported."
+            )
+
+        if user_data.role == Role.STUDENT:
+            return (
+                API_URL_STUDENTS_LIST_CREATE,
+                user_data.student_as_dict(update=update, personal_info_id=personal_info_id),
+            )
+
+        if user_data.role == Role.TEACHER and user_data.teacher_is_under_18:
+            return (
+                API_URL_YOUNG_TEACHERS_LIST_CREATE,
+                user_data.teacher_under_18_as_dict(
+                    update=update, personal_info_id=personal_info_id
+                ),
+            )
+
+        return (
+            API_URL_TEACHERS_LIST_CREATE,
+            user_data.teacher_as_dict(update=update, personal_info_id=personal_info_id),
+        )
+
+    @staticmethod
+    def _generate_failure_message_for_person_creation(
+        context: CUSTOM_CONTEXT_TYPES, personal_info_id: int
+    ) -> str:
+        user_data = context.user_data
+        teacher_age_infix = (
+            "young " if user_data.role == Role.TEACHER and user_data.teacher_is_under_18 else ""
+        )
+        return f"Failed to create {teacher_age_infix}{user_data.role}, ID {personal_info_id}"
+
+    @staticmethod
+    def _generate_success_message_for_person_creation(
+        context: CUSTOM_CONTEXT_TYPES, personal_info_id: int
+    ) -> str:
+        """Generates success message in Markdown V2 after a student/teacher was created."""
+        user_data = context.user_data
+
+        teacher_age_infix = (
+            "young " if user_data.role == Role.TEACHER and user_data.teacher_is_under_18 else ""
+        )
+
+        if user_data.role == Role.TEACHER:
+            url_prefix = (
+                API_URL_YOUNG_TEACHER_RETRIEVE
+                if user_data.teacher_is_under_18
+                else API_URL_TEACHER_RETRIEVE
+            )
+        elif user_data == Role.STUDENT:
+            url_prefix = API_URL_STUDENT_RETRIEVE
+        else:
+            raise NotImplementedError(f"{user_data.role=} not supported")
+
+        success_message = (
+            f"Created {teacher_age_infix}{user_data.role} "
+            f"[{escape_for_markdown(user_data.first_name)} "
+            f"{escape_for_markdown(user_data.last_name)}]"
+            f"({url_prefix}{personal_info_id}), ID {personal_info_id}\\."
+        )
+        if user_data.role == Role.TEACHER and user_data.teacher_can_host_speaking_club:
+            success_message += (
+                f" @{os.environ.get('SPEAKING_CLUB_COORDINATOR_USERNAME')} "
```

Comment:
Let's put it to some config/constants file and use it as constant, without reading from envs for every call

Reply:
Good point. Done